app/stores/ada/YoroiTransferStore.js

```diff
@@ -0,0 +1,201 @@
+  @observable error: ?LocalizableError = null;
+  @observable transferTx: ?TransferTx = null;
+
+  _errorWrapper = <PT, RT>(func: PT=>Promise<RT>): (PT => Promise<RT>) => (async (payload) => {
+    try {
+      return await func(payload);
+    } catch (error) {
+      Logger.error(`YoroiTransferStore ${stringifyError(error)}`);
+      runInAction(() => {
+        this.status = 'error';
+        this.error = localizedError(error);
+      });
+      throw error;
+    }
+  });
+
+  setup(): void {
+    const actions = this.actions.ada.yoroiTransfer;
+    actions.setupTransferFundsWithMnemonic.listen(
+      this._errorWrapper(this._setupTransferFundsWithMnemonic)
+    );
+    actions.backToUninitialized.listen(this._backToUninitialized);
+    actions.transferFunds.listen(this._errorWrapper(this._transferFunds));
+    actions.cancelTransferFunds.listen(this._reset);
+  }
+
+  teardown(): void {
+    super.teardown();
+    this._reset();
+  }
+
+  _setupTransferFundsWithMnemonic = async (payload: { recoveryPhrase: string }): Promise<void> => {
+    this._updateStatus('checkingAddresses');
+
+    const { masterKey, addresses } = await this._restoreWalletForTransfer(payload.recoveryPhrase);
+
+    this._updateStatus('generatingTx');
+
+    const cryptoWallet = getCryptoWalletFromMasterKey(masterKey, '');
+    const addressKeys = {};
+    addresses.forEach(({ address, accountIndex, addressType, index }) => {
+      const account = cryptoWallet.bip44_account(
+        RustModule.Wallet.AccountIndex.new(accountIndex | HARD_DERIVATION_START)
+      );
+      const chainPrv = account.bip44_chain(addressType === 'Interanl');
+      const keyPrv = chainPrv.address_key(RustModule.Wallet.AddressKeyIndex.new(index));
+      addressKeys[address] = keyPrv;
+    });
+
+    const outputAddr = await getReceiverAddress();
+    // Possible exception: NotEnoughMoneyToSendError
+    const transferTx = await generateTransferTx({
+      outputAddr,
+      addressKeys,
+      getUTXOsForAddresses:
+        this.stores.substores.ada.stateFetchStore.fetcher.getUTXOsForAddresses,
+    }, true);
+
+    runInAction(() => {
+      this.transferTx = transferTx;
+    });
+
+    this._updateStatus('readyToTransfer');
+  }
+
+  _backToUninitialized = (): void => {
+    this._updateStatus('uninitialized');
+  }
+
+  /** Updates the status that we show to the user as transfer progresses */
+  @action.bound
+  _updateStatus(s: TransferStatus): void {
+    this.status = s;
+  }
+
+  /** Broadcast the transfer transaction if one exists and proceed to continuation */
+  _transferFunds = async (payload: {
+    next: void => void
+  }): Promise<void> => {
+    try {
+      const { next } = payload;
+
+      if (!this.transferTx) {
+        throw new NoTransferTxError();
+      }
+      await this.transferFundsRequest.execute({
+        signedTx: this.transferTx.signedTx
+      });
+      // TBD: why do we need a continuation instead of just pustting the code here directly?
```

Comment:
TBD? I agree -- the continuation here doesn't make sense to me either.

Reply:
Copied from DaedalusTransferStore.js. Maybe this is because we want to call the continuation function before calling _reset()? And what `next` does actually belongs to the view (YoroiTransferPage.js) instead of state (the store)?